presto-hive/src/main/java/com/facebook/presto/hive/metastore/CachingHiveMetastore.java

```diff
@@ -82,14 +83,22 @@ public CachingHiveMetastore(@ForCachingHiveMetastore ExtendedHiveMetastore deleg
 
     public CachingHiveMetastore(ExtendedHiveMetastore delegate, ExecutorService executor, Duration cacheTtl, Duration refreshInterval)
```

Comment:
Make this constructor a factory method so that it can skip creation of the caching layer entirely if `cacheTtl` is 0.